span.go

```diff
@@ -119,6 +120,34 @@ func (s *span) setTracerTags(tags []tag) {
 	s.Unlock()
 }
 
+func (s *span) LogFields(fields ...log.Field) {
+	s.Lock()
+	defer s.Unlock()
+	if !s.context.IsSampled() {
```

Comment:
It looks like all the places that call this function check that the span whether the span is sampled before the execution of this function. Do we want to be checking twice? Can a span be sampled midway through somehow?